packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
+    value,
+    ...customProps
+  } = props;
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const momentDateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormatOrder = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateSeparator = intl.formatMessage({ id: 'Terra.datePicker.separator' });
+  const label = ariaLabel || intl.formatMessage({ id: 'Terra.datePicker.date' });
+  const placeholderValues = DateUtil.getDateInputValues(dateFormatOrder, placeholder, dateSeparator);
+  const theme = React.useContext(ThemeContext);
+
+  const setDayRef = useCallback(node => {
+    dayInputRef.current = node;
+    if (inputRefCallback && dateFormatOrder === 'YMD') {
+      inputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, inputRefCallback]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  const setMonthRef = useCallback(node => {
+    monthInputRef.current = node;
+  }, []);
 
-    if (!attributes.readOnly && this.onCalendarButtonClick && this.props.onClick) {
-      this.onCalendarButtonClick(event, this.props.onClick);
+  const setYearRef = useCallback(node => {
+    yearInputRef.current = node;
+    if (inputRefCallback && (dateFormatOrder === 'MDY' || dateFormatOrder === 'DMY')) {
+      inputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, inputRefCallback]);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
+    }
+  }, [shouldShowPicker, onClick]);
+
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, momentDateFormat)) {
+      const parseDate = DateUtil.getDateInputValues('YMD', dateValue, '-');
```

Comment:
Why is this separator hard-coded to `-`? Should it be the following?
```suggestion
      const parseDate = DateUtil.getDateInputValues('YMD', dateValue, dateSeparator);
```

Reply:
The reason the separator is hard coded here is because the dateValue is in ISO format as seen [here](https://github.com/cerner/terra-framework/blob/ba9d3c9b977ded21c1db0789cbe2b1e43dd3c7c6/packages/terra-date-picker/src/DateInput.jsx#L174).